src/System.Management.Automation/engine/remoting/fanin/OutOfProcTransportManager.cs

```diff
@@ -1172,6 +1159,54 @@ internal override void CreateAsync()
             SendOneItem();
         }
 
+        private void StartRedirectionReaderThreads(Process serverProcess)
+        {
+            Thread outputThread = new Thread(ProcessOutputData);
+            outputThread.IsBackground = true;
+            outputThread.Name = "Out-of-Proc Job Transport Output Thread";
+
+            Thread errorThread = new Thread(ProcessErrorData);
+            errorThread.IsBackground = true;
+            errorThread.Name = "Out-of-Proc Job Transport Error Thread";
+
+            outputThread.Start(serverProcess.StandardOutput);
+            errorThread.Start(serverProcess.StandardError);
+        }
+
+        private void ProcessOutputData(object arg)
+        {
+            if (arg is StreamReader reader)
+            {
+                string data = reader.ReadLine();
```

Comment:
One concern I have is if the host process exits suddenly.  We want to make sure that any exception the StreamReader throws is caught and reported, so that the client doesn't crash.  Please see SSH and NamedPipe transports.  We may need try/catch, something like:
```
catch (ObjectDisposedException)
            {
                // Normal reader thread end.
            }
            catch (Exception e)
            {
                string errorMsg = (e.Message != null) ? e.Message : string.Empty;
                _tracer.WriteMessage("SSHClientSessionTransportManager", "ProcessErrorThread", Guid.Empty,
                    "Transport manager error thread ended with error: {0}", errorMsg);

                PSRemotingTransportException psrte = new PSRemotingTransportException(
                    StringUtil.Format(RemotingErrorIdStrings.SSHClientEndWithErrorMessage, errorMsg),
                    e);
                HandleSSHError(psrte);
            }
```

Reply:
`try/catch` was added to the output/error reader threads. Thanks @PaulHigin!

Here we do the same as what `Process.BeginOutputReadLine()` and `Process.BeginErrorReadLine()` do -- treat `IOException` as EOF.